### PR TITLE
Reorder JSON to be more consistent

### PR DESCRIPTION
**Pull request by SPConrad**
Increase width of cells to be more like real dashboards
Consolidate `Job` and `Widget` select boxes as they must be the same
Add Name Label to cells, determinative logic to display name/title/missing
Fix numberOfColumns to be int instead of string

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -362,12 +362,10 @@ function addWidgetsFromJSON(jsonObject, numberOfColumns, numberOfRows) {
         const widgetWidth = widget.width;
         const widgetHeight = widget.height;
         let key = widgetCol + "_" + widgetRow;
-        // console.log("REGULAR KEY : " + key);
```
Comment (iamsellek) [inline main.js]: 👍

File: main.js [hunk 2]
```diff
@@ -379,40 +377,52 @@ function addWidgetsFromJSON(jsonObject, numberOfColumns, numberOfRows) {
...
-                const id = addCell();
-                makeHiddenCell(document.getElementById(id));
-                // console.log("HIDDEN KEY : " + key);
+
+        if (widget == undefined) {
+            // widget is not in the available list, hide
+            const id = addCell();
+            makeHiddenCell(document.getElementById("widget-box-" + id));
+        } else {
+            if (jsonObject.config[widget.config].widgetTitle !== undefined) {
```
Comment (iamsellek) [inline main.js]: If there's any chance that `jsonObject.config[widget.config]` will not exist here, you'll probably want to add a check for that first and `&&` it with this current check.

File: main.js [hunk 3]
```diff
@@ -379,40 +377,52 @@ function addWidgetsFromJSON(jsonObject, numberOfColumns, numberOfRows) {
...
+
+            let widgetNameLabel = document.getElementById("name-label-" + widgetID);
+            let widgetConfigLabel = document.getElementById("config-label-" + widgetID);
+            let selectedWidget = getSelectListElement(widgetsListName, widgetID);
+
+            widgetNameLabel.innerText = name;
+            widgetConfigLabel.innerText = widgetConfig
+            selectedWidget.value = widgetWidget;
+            
+            if (widgetConfig) {
```
Comment (iamsellek) [inline main.js]: On line 417, you're setting widgetConfigLabel to widgetConfig and then you're checking for it's existence here. Should 417 move into this if statemnt?

File: main.js [hunk 4]
```diff
@@ -474,15 +484,16 @@ function generateJSON() {
...
 
-        if (!selectedJob || !selectedWidget || selectedJob === jobsPlaceholderMessage || selectedWidget === widgetsPlaceholderMessage) {
+        if (!widgetNameLabel || !selectedWidget || widgetNameLabel === jobsPlaceholderMessage || selectedWidget === widgetsPlaceholderMessage) {
             //Require that a job and a widget be selected to include this cell as part of the generated JSON
             continue;
         }
 
-        const configJSONName = selectedJob ? (selectedJob + "_" + itemID) : "";
+        const configJSONName = getItemConfigName(itemID);
+        //widgetNameLabel ? (widgetNameLabel + "_" + itemID) : "";
```
Comment (iamsellek) [inline main.js]: Should this still be here?

File: main.js [hunk 5]
```diff
@@ -543,7 +554,9 @@ function parseJSONString(jsonString, column, row) {
 
 //Generates the HTML for a cell in the grid
 function generateItemDOMFragment(id) {
-    let jobsDropDownHtml = createDropDownHTML(jobsListName, jobsList, id, jobsPlaceholderMessage);
+    // TODO: variable-ify these strings
```
Comment (iamsellek) [inline main.js]: Does this TODO need to be completed before this PR goes in?

File: main.js [hunk 6]
```diff
@@ -562,6 +576,11 @@ function generateItemDOMFragment(id) {
     return fragment;
 }
 
+function createLabelHTML(type, itemId, placeholderMessage) {
+    let html = '<p class="cell-' + type + '-label" id="' + type + '-label-' + itemId + '"> ' + placeholderMessage + '</p>'
```
Comment (iamsellek) [inline main.js]: Can you use ticks here? (Top left of keyboard). They tend to make stuff like this a ton easier to read. [Check here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) for more info.

File: main.js [hunk 7]
```diff
@@ -710,8 +729,10 @@ function configEditedListener(configJsonDisplayTextArea) {
 //Listener that handles when the use selects a new job or widget, and pre-fills the config textfield with a template based on the item selected
 function handleListChangeEvent(selectElement, id) {
 
+    /// all of the jobs have the same name as the widgets, just use the widget name
```
Comment (iamsellek) [inline main.js]: 👍

**Review by iamsellek — CHANGES_REQUESTED** [review]

File: main.js [hunk 8]
```diff
@@ -379,40 +377,51 @@ function addWidgetsFromJSON(jsonObject, numberOfColumns, numberOfRows) {
...
         const key = column + "_" + row;
         const widget = widgetsDictionary[key];
-        if (!widget || existingWidgets.indexOf(widget) >= 0) {
-            //no widget found for this part of the grid, add an empty cell instead
-            if (!widget && existingWidgets.indexOf(widget) < 0) {
-                const id = addCell();
-                makeHiddenCell(document.getElementById(id));
-                // console.log("HIDDEN KEY : " + key);
+
+        if (widget == undefined || jsonObject.config[widget.config]) {
```
Comment (iamsellek) [inline main.js]: I think you want a not in front of jsonObject here.

**Review by iamsellek — CHANGES_REQUESTED** [review]

File: main.js (hunk 5 above)
Comment (SPConrad) [reply main.js]: Nope

File: main.js (hunk 4 above)
Comment (SPConrad) [reply main.js]: Nope

File: main.js (hunk 3 above)
Comment (SPConrad) [reply main.js]: Yes

File: main.js (hunk 2 above)
Comment (SPConrad) [reply main.js]: Yep

File: main.js (hunk 8 above)
Comment (SPConrad) [reply main.js]: 🤦‍♂️

**Review by iamsellek — APPROVED** [review]

**Pull request merged**